scripts/barclamp_install.rb

```diff
@@ -97,97 +115,34 @@ def usage()
     next
   end
 
-  # We assume the barclamp and crowbar keys exist and their values are hashes.
+  # We assume the barclamp key exists and its value is a hash.
   version = (barclamp["barclamp"]["version"] || 0).to_i
-  order   = (barclamp["crowbar"]["order"] || 9999).to_i
+  if barclamp["nav"] && version < 1
+    puts "Barclamp at #{yml_file} has incompatible navigation, skipping"
+    next
+  end
 
-  barclamps[name] = { :src => File.dirname(yml_file), :name => name, :order => order, :yaml => barclamp, :version => version }
-  debug "barclamp[#{name}] = #{barclamps[name].pretty_inspect}"
+  barclamps[name] = { :name => name, :migrate => do_chef && check_schema_migration(name) }
```

Comment:
Style/HashSyntax: Use the new Ruby 1.9 hash syntax. (https://github.com/bbatsov/ruby-style-guide#hash-literals)